### PR TITLE
chore(docs): Azure App Insights Recipes

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the missing recipe documentation for the azure app insights integration

## Changes

- Adds recipes
- Adds links to side bars
- Adds links to overview

## Fixes

- #2820 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

Screenshot 2023-06-26 at 13 13 58

### CODE REVIEW

File: docs/docs/configuration/connecting-to-data-stores/azure-app-insights.md [hunk 1]
```diff
@@ -45,17 +45,16 @@ Run this command in the terminal and specify the file above.
...
+To learn more, [read the recipe on running a sample app with Azure App Insights and Tracetest](../../examples-tutorials/recipes/running-tracetest-with-azure-app-insights).
+:::
 
 ## OpenTelemetry Collector
 
 You can configure Tracetest to listen for incoming telemetry data from ports `4317` and `4318` for gRPC and REST accordingly, giving you the option to stream the information to both Azure App Insights and Tracetest at the same time.
 
 :::tip
-Need help configuring the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector-contrib) to send trace data from your application to Azure App Insights? Read more in [the reference page here](../opentelemetry-collector-configuration-file-reference).
+Need help configuring the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector-contrib) so send trace data from your application to Azure App Insights? Read more in [the reference page here](../opentelemetry-collector-configuration-file-reference).
```
Comment (jfermi) [inline docs/docs/configuration/connecting-to-data-stores/azure-app-insights.md:57]:
```suggestion
Need help configuring the [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector-contrib) to send trace data from your application to Azure App Insights? Read more in [the reference page here](../opentelemetry-collector-configuration-file-reference).
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights-collector.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+All `services` in the `docker-compose.yaml` are on the same network and will be reachable by hostname from within other services.
+
+## Node.js App
+
+The Node.js app is a simple Express app, contained in the `src/index.js` file.
+
+It is instrumented using the [Official OpenTelemetry Node.js SDK](https://www.npmjs.com/package/@opentelemetry/sdk-node) wrapping the application code to send telemetry data to the OpenTelemetry Collector.
+
+The key following is the instrumentation section from the `src/tracing.js` file.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights-collector.md:48]:
```suggestion
The following key is the instrumentation section from the `src/tracing.js` file.
```
Is 'key' the correct word or should it be 'code'?

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights-pokeshop.md [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+  globalTracer = tracer;
+
+  return globalTracer;
+}
+```
+
+The `docker-compose.yaml` file includes the definitions for all of the required services by the Pokeshop API, which includes:
+
+- **Postgres** - To save Pokemon information.
+- **Redis** - For in memory strage.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights-pokeshop.md:110]:
```suggestion
- **Redis** - For in memory storage.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+All `services` in the `docker-compose.yaml` are on the same network and will be reachable by hostname from within other services.
+
+## Node.js App
+
+The Node.js app is a simple Express app, contained in the `src/index.js` file.
+
+It is instrumented using the [applicationinsights SDK](https://www.npmjs.com/package/applicationinsights) wrapping the application code to send telemetry data directly to the Azure cloud.
+
+The key following is the instrumentation section from the `src/tracing.js` file.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights.md:46]:
```suggestion
The following key is the instrumentation section from the `src/tracing.js` file.
```
Is 'key' the correct word or should it be 'code'?

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+RUN npm install
+COPY ./src .
+
+EXPOSE 3000
+CMD [ "npm", "start" ]
+```
+
+## Tracetest
+
+The `docker-compose.yaml` includes three other services.
```
Comment (jfermi) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights.md:91]: I only see two other services :)

**Review by jfermi — COMMENTED** [review]
Some editing suggestions.

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights-collector.md (hunk 2 above)
Comment (adnanrahic) [reply docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights-collector.md:48]: I'd say we should use "code".

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights-collector.md [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+
+And the [App Insights Instrumentation Key](https://learn.microsoft.com/en-us/azure/bot-service/bot-service-resources-app-insights-keys?view=azure-bot-service-4.0) from your instance.
+
+## Project Structure
+
+The project is built with Docker Compose.
+
+### 1. Node.js App
+
+The `Dockerfile` in the root directory is for the Node.js app.
```
Comment (adnanrahic) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights-collector.md:32]:
```suggestion
The `Dockerfile` and the `docker-compose.yaml` in the root directory are for the Node.js app.
```

File: docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights-collector.md [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+
+### 2. Tracetest
+
+The `docker-compose.yaml` file, `tracetest.provision.yaml`, and `tracetest-config.yaml` in the `root` directory are for the setting up the Node.js App and Tracetest.
```
Comment (adnanrahic) [inline docs/docs/examples-tutorials/recipes/running-tracetest-with-azure-app-insights-collector.md:36]:
```suggestion
The `docker-compose.yaml` file, `tracetest.provision.yaml`, and `tracetest-config.yaml` in the `tracetest` directory are for setting up the Node.js App and Tracetest.
```

**Review by adnanrahic — APPROVED** [review]

**Pull request merged**